components/app_bar/app_bar.tsx

```diff
@@ -24,26 +25,32 @@ export default function AppBar() {
         return null;
     }
 
-    // Order the plugin components so that Playbooks is the first one,
-    // Boards is the second one, and the others remain in the same place.
-    const orderedAppBarPluginComponents = [...appBarPluginComponents];
-    orderedAppBarPluginComponents.sort((a, b) => {
-        switch (a.pluginId) {
-        case 'playbooks':
-            return -1;
+    const coreProductsIds = ['playbooks', 'focalboard'];
 
-        case 'focalboard':
-            return b.pluginId === 'playbooks' ? 1 : -1;
+    // The type guard in the filter (which removes all undefined elements) is needed for
+    // Typescript to correctly type coreProducts.
+    const coreProducts = coreProductsIds.
+        map((id) => appBarPluginComponents.find((element) => element.pluginId === id)).
+        filter((element): element is PluginComponent => Boolean(element));
```

Comment:
Huh I didn't know about the `is` keyword. Is there an equivalent that I may be familiar with, like `as unknown as PluginComponent`? Not saying this needs to change, just trying to understand what's going on there.

Reply:
The function passed to `filter` is a type guard, and the `is` keyword creates a type predicate: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards
At the beginning I just did `filter((element) => Boolean(element))`, but then Typescript complained that `coreProducts` was a `(PluginComponent | undefined)[]` instead of a `PluginComponent[]`. It just needs the type guard to correctly type `coreProducts` :man_shrugging: 
(We could have also done `as PluginComponent[]` at the end of everything, though)